amazon_transcribe/serialize.py

```diff
@@ -25,62 +24,118 @@
     BaseEvent,
 )
 
-HEADER_VALUE = Union[int, None, str]
 
-
-class Serializer:
-    def __init__(self):
-        raise NotImplementedError("Serializer")
-
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        """Serialize out to payload and headers."""
-        raise NotImplementedError("serialize")
-
-    def serialize_to_request(self) -> PreparedRequest:
-        """Serialize parameters into an HTTP request."""
-        raise NotImplementedError("serialize_to_request")
-
-
-class TranscribeStreamingRequestSerializer(Serializer):
+class TranscribeStreamingSerializer:
     """Convert StartStreamTranscriptionRequest into a
     Request object for streaming to the Transcribe service.
     """
 
-    def __init__(self, endpoint, transcribe_request):
-        self.endpoint: str = endpoint
-        self.method: str = "POST"
-        self.request_uri: str = "/stream-transcription"
-        self.request_shape: StartStreamTranscriptionRequest = transcribe_request
-
-    def serialize(self) -> Tuple[Dict[str, HEADER_VALUE], BufferedIOBase]:
-        headers = {
-            "x-amzn-transcribe-language-code": self.request_shape.language_code,
-            "x-amzn-transcribe-sample-rate": self.request_shape.media_sample_rate_hz,
-            "x-amzn-transcribe-media-encoding": self.request_shape.media_encoding,
-            "x-amzn-transcribe-vocabulary-name": self.request_shape.vocabulary_name,
-            "x-amzn-transcribe-session-id": self.request_shape.session_id,
-            "x-amzn-transcribe-vocabulary-filter-method": self.request_shape.vocab_filter_method,
-            "x-amzn-transcribe-vocabulary-filter-name": self.request_shape.vocab_filter_name,
-            "x-amzn-transcribe-show-speaker-label": self.request_shape.show_speaker_label,
-            "x-amzn-transcribe-enable-channel-identification": self.request_shape.enable_channel_identification,
-            "x-amzn-transcribe-number-of-channels": self.request_shape.number_of_channels,
-        }
+    def _serialize_str_header(self, header: str, value: Optional[str]):
+        if value is None:
+            return {}
+        elif not isinstance(value, str):
+            raise SerializerException(
+                f'Unexpected param type encountered: "{type(value)}"'
+            )
+        else:
+            return {header: str(value)}
+
+    def _serialize_int_header(self, header: str, value: Optional[int]):
+        if value is None:
+            return {}
+        elif not isinstance(value, int):
+            raise SerializerException(
+                f'Unexpected param type encountered: "{type(value)}"'
+            )
+        else:
+            return {header: str(value)}
```

Comment:
Another thought here to help simplify things, all of the standard headers begin with `x-amzn-transcribe`, we could potentially add a header-prefix that defaults to that value. That gives us two advantages, by allowing the `serialize_start_stream_transcription_request` to be a bit more compact and giving us the actual field name (e.g. `language-code`) to use in the exception message.